src/utils/typedData/types.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Infer,
+  array,
+  intersection,
+  number,
+  object,
+  optional,
+  record,
+  refine,
+  string,
+  type as t,
+  union,
+} from 'superstruct';
+
+export const STATIC_TYPES = ['felt', 'felt*'];
```

Comment:
I understand what you are doing but I find it weird to have arrays (i.e. `felt*`) listed as static types while they are dynamic variables like structs. Maybe it's just the naming.